Tests/PostgresNIOTests/New/PostgresConnectionTests.swift

```diff
@@ -275,6 +275,100 @@ class PostgresConnectionTests: XCTestCase {
         }
     }
 
+    struct TestPrepareStatement: PostgresPreparedStatement {
+        static var sql = "SELECT datname FROM pg_stat_activity WHERE state = $1"
+        typealias Row = String
+
+        var state: String
+
+        func makeBindings() -> PostgresBindings {
+            var bindings = PostgresBindings()
+            bindings.append(.init(string: self.state))
+            return bindings
+        }
+
+        func decodeRow(_ row: PostgresNIO.PostgresRow) throws -> Row {
+            try row.decode(Row.self)
+        }
+    }
+
+    func testPreparedStatement() async throws {
```

Comment:
I know that these tests are somewhat painful to write but we should embrace them, as we can produce hard to reproduce scenarios here. 

For this reason, please lets try to create those scenarios in extra tests here:

- Sending the a prepared query twice, before the preparation step has succeeded. Is the order of the send statements still correct? Is the query just prepared once?
- Sending the a prepared, after the the query of the same type has already succeeded. Is the query executed right away?
- Failing a prepared statement in preparation are other executes failed in the same way right away?

Happy to help.